tests/test_gates.py

```diff
@@ -138,6 +138,42 @@ def output(input):
     assert np.allclose(unitary, unitary_expected)
 
 
+@pytest.mark.parametrize("wires, pauli_words", [[4, "XYIZ"], [3, {"X", "I", "Z"}], [1, "X"]])
+def test_paulirot_unitary_correct(wires, pauli_words):
+    """Test if lightning.gpu correctly applies PauliRot by reconstructing the unitary matrix and
+    comparing to the expected version"""
+
+    dev = qml.device("lightning.gpu", wires=wires)
+    p = [0.1]
+
+    @qml.qnode(dev)
+    def output(input):
+        qml.BasisState(input, wires=range(wires))
+        qml.PauliRot(*p, pauli_words, wires=range(wires))
+        return qml.state()
+
+    unitary = np.zeros((2**wires, 2**wires), dtype=np.complex128)
+
+    for i, input in enumerate(itertools.product([0, 1], repeat=wires)):
+        out = output(input)
+        unitary[:, i] = out
+
+    unitary_expected = qml.matrix(qml.PauliRot(*p, pauli_words, wires=range(wires)))
+
+    assert np.allclose(unitary, unitary_expected)
+
+
+@pytest.mark.parametrize("wires, pauli_words", [[4, "XYIW"], [1, {"Q"}]])
```

Comment:
what about switch {"Q"} to "Q"?

Reply:
This one is just testing different input types. I added a new test case for `[1, "Q"]`.